api/src/test/java/uk/gov/dhsc/htbhf/claimant/reporting/GoogleAnalyticsClientTest.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.dhsc.htbhf.claimant.reporting;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import io.zonky.test.db.AutoConfigureEmbeddedDatabase;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock;
+import uk.gov.dhsc.htbhf.claimant.exception.GoogleAnalyticsException;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowableOfType;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.springframework.http.MediaType.TEXT_PLAIN_VALUE;
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@AutoConfigureEmbeddedDatabase
+@AutoConfigureWireMock(port = 8120)
+class GoogleAnalyticsClientTest {
+
+    private static final String REPORT_ENDPOINT = "/collect";
+    // properties need to be ordered for the assertion against the encoded string to work
+    private static final Map<String, String> REPORT_PROPERTIES = new LinkedHashMap<>();
+    // google analytics expects the payload to be url encoded and in the format of a query string
+    private static final String URL_ENCODED_REPORT_PROPERTIES = "cd8=Bristol+West&cd9=NHS+Bristol%2C+North+Somerset+and+South+Gloucestershire";
+
+    static {
+        REPORT_PROPERTIES.put("cd8", "Bristol West");
+        REPORT_PROPERTIES.put("cd9", "NHS Bristol, North Somerset and South Gloucestershire");
+    }
+
+    @Value("${google-analytics.base-uri}")
+    private String baseUri;
```

Comment:
might be worth a comment explaining that this is set to localhost in test/resources/application.yml

Reply:
Not sure if that's necessary as we haven't done so for any of the other properties set to localhost.